model/src/main/java/org/cbioportal/model/ClinicalDataModel.java

```diff
@@ -0,0 +1,50 @@
+package org.cbioportal.model;
```

Comment:
This was a question I brought up in one of the first meeting. Spark Encoders can only map Rows to POJOs and ClinicalData is not a POJO, since it extends UniqueKeyBase and is also nested, containing ClinicalAttribute object as an attribute. I created a POJO model for this method, because we conditionally join mutation and fraction data and with join the order is not preserved, so row.get(index) doesn't work.

Reply:
Does the current implementation return a unique sample/patient key?

Reply:
i think the unique keys get populated, but in this case, the Study view APIs use ClinicalData to create lists of ClinicalDataCountItem or DataBin.